server/src/server.ts

```diff
@@ -105,30 +107,58 @@ connection.onInitialize(
 );
 
 connection.onInitialized((_params) => {
-    connection.workspace.getConfiguration('yseopml.parseAllProjectFilesAtStartup').then((_confValue) => {
-        parseAllProjectFilesAtStartup = _confValue;
-        if (!parseAllProjectFilesAtStartup) {
-            connection.console.log('Not parsing project files at startup.');
-            return;
-        }
-        connection.console.log('Parsing project files at startup.');
-        connection.workspace.getWorkspaceFolders().then((_value) => {
-            const workspacePath = url.fileURLToPath(_value[0].uri);
-            glob('**/project.kao', {
-                cwd: workspacePath,
-            })
-                .then((_matches) => {
-                    for (const filePath of _matches) {
-                        if (openProjectFile(workspacePath, path.join(workspacePath, filePath))) {
-                            // There should be only one `project.kao` file. If we found a good candidate, stop the loop.
-                            return;
+    connection.workspace
+        .getConfiguration('yseopml.kaoFiles')
+        .then((_kaoFiles) => {
+            // initialize array kaoFilesToParse
+            kaoFilesToParse = _kaoFiles;
+        })
+        .then(() => {
```

Comment:
Can’t say I’m thrilled by this 35+-line then-lambda. xD Wow it even contains another huge then-lambda.
Can you take a couple of minutes to see if there is stuff in there that could be extracted to methods without too much hassle?